libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/NavRouteOptionsBuilder.kt

```diff
@@ -0,0 +1,75 @@
+package com.mapbox.navigation.core.routeoptions
+
+import android.content.Context
+import com.mapbox.api.directions.v5.DirectionsCriteria
+import com.mapbox.api.directions.v5.models.Bearing
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.extensions.applyDefaultNavigationOptions
+import com.mapbox.navigation.base.extensions.applyLanguageAndVoiceUnitOptions
+
+class NavRouteOptionsBuilder internal constructor(
+    private val locationProvider: LocationProvider
+) {
+
+    private lateinit var destination: Point
+    private var destinationName: String? = null
+    private val builder = RouteOptions.builder()
+
+    fun fromCurrentPosition(): NavRouteOptionsBuilder {
```

Comment:
This method is supposed to add all the recommended fields to request: coordinate, bearing, z level